input-output/pydicom_series.py

```diff
@@ -598,10 +612,8 @@ def _finish(self):
             # Set attributes
             ds = self._datasets[0]
             self._info = self._datasets[0]
-            self._shape = [ds.Rows, ds.Columns]
-            self._sampling = [
-                float(ds.PixelSpacing[0]), float(ds.PixelSpacing[1])
-            ]
+            self._shape = find_shape(ds)
+            self._sampling = float(ds.PixelSpacing[0]), float(ds.PixelSpacing[1])
```

Comment:
I didn't notice this before, but now wondering about `sampling` and `_sampling` - seems an odd name given other meaning in `SamplesPerPixel`.  I imagine this came through from the original contribution, but perhaps we can change to just use `ds.PixelSpacing` which is already a tuple and should already be float values?

Reply:
Yes, I'll change the names from `sampling` to `spacing`.
`ds.PixelSpacing` is actually of a specific pydicom type - `MultiValue` - not a `tuple`, which is a bit harder to work with. How about:
```python
self._spacing = tuple(ds.PixelSpacing)
```
?

Reply:
It shouldn't be harder to work with - it is derived from Python's `MutableSequence` (basically `list`), so you can do everything a tuple can, but also reassign values within it.  It can also just compare the two directly for equality rather than worrying about individual values.

```python
if ds1.PixelSpacing != ds2.PixelSpacing:
   ...
```

I think that is a lot easier to understand.

Reply:
I am not worried about the comparison, what worries me more is this line:
https://github.com/pydicom/contrib-pydicom/blob/938755515445dfdcf3ea22ee526017fb2ced3232/input-output/pydicom_series.py#L676
Actually, I think it's better to stick with the current version of this so that `_sampling` (`_spacing`) will contain elements of the same type

Reply:
Ok, we can use
```python 
sampling.insert(0, distance_mean)
self._sampling = sampling
```
I prefer the current version, but it's up to you

Reply:
> what worries me more is this line:
> https://github.com/pydicom/contrib-pydicom/blob/938755515445dfdcf3ea22ee526017fb2ced3232/input-output/pydicom_series.py#L676

Is that actually used anywhere?  I just searched on the webpage, it is stored and returned in the property method, but not otherwise used (one place using the property is commented out).

Reply:
Well, maybe in this script not, but when you add multiple slices together the spacing (including between the slices) is an interesting property.

Reply:
Yeah, I suppose.  It doesn't really matter too much, I'm just being a bit picky today.  Perhaps pandemic restrictions fatigue.

Whatever you think is easiest at this point is fine with me.  I'd prefer original DICOM terms as much as is as relatively easy to do, but we don't need to tear apart the whole code.